src/prng/mod.rs

```diff
@@ -202,31 +190,26 @@
 //! From the context of any PRNG, one can ask the question *given some previous
 //! output from the PRNG, is it possible to predict the next output value?*
 //!
-//! Simple PRNGs tend to fall into one of two categories here; *yes* and
-//! *possibly*. In some cases prediction is trivial; e.g. plain Xorshift outputs
-//! part of its state without mutation, and prediction is as simple as seeding
-//! a new Xorshift generator from four `u32` outputs. The widely-used Mersenne
-//! Twister algorithms are also easy to predict, though more samples are
-//! required (624 `u32` samples, in the case of MT19937). At the other end of
-//! the range, the PCG generators are small and fast yet designed to be
-//! [hard to predict](http://www.pcg-random.org/predictability.html),
-//! however they should not be trusted to be secure.
-//!
-//! The basic security that CSPRNGs must provide is precisely the above, i.e.
-//! infeasible to predict. This requirement is formalised as the
-//! [next-bit test]; this is roughly stated as: given the first *k* bits of a
-//! random sequence, the sequence satisfies the next-bit test if there is no
-//! algorithm able to predict the next bit using reasonable computing power.
-//! We therefore rate predictability as one of *secure*, *unknown* (meaning
-//! apparently strong and no known attacks, but not trusted) or *insecure*.
-//!
-//! A further security that *some* CSPRNGs provide is forward-security: in the
-//! event that the CSPRNGs state is revealed after generating some output, it
-//! must be infeasible to reconstruct previous states or output. Note that many
-//! CSPRNGs *do not* have forward-security in their usual formulations. It is
-//! possible to make any CSPRNG forward-secure by reseeding it from its own
-//! output after generating each output value, however this is quite slow
-//! (especially since CSPRNGs usually cache a block of output values).
+//! Normal PRNGs tend to fall into one of two categories here; *yes* and
+//! *with effort*. In some cases prediction is trivial; e.g. plain Xorshift
```

Comment:
"with effort" is not a good description in my opinion; it implies these are all breakable with reasonable computing power. In reality that may not be the case for all of them, or at least we don't currently know how to do so. Read this section: http://www.pcg-random.org/predictability.html#predictability-of-the-pcg-family

Or have you already cracked PCG?

Fundamentally non-crypto and CS-PRNGs are not that different; it's just that CSPRNGs use *proven secure* transformations or many more operations than required for good bit-mixing.

Reply:
I strongly disagree. I don't think the distinction between between "with effort" and "no" has merit. Either it is a CSPRNG and should be advertized as such, or it is not.

Crypto is notoriously hard, and "the author didn't break it" or "you didn't break it" does not cut it. The bar is much higher and requires cryptanalysis by several experts, which did not happen for the non-CSPRNGs.

Note that the claims about predictability are one of the biggest criticisms of the PCG paper.

Reply:
> Or have you already cracked PCG?

It already was, without much effort

Reply:
Finally found it (posting here so I can find it back later :smile:): https://github.com/mariuslp/PCG_attack